main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -90,16 +90,25 @@ object Resolver extends Phase[NamedAst.Root, ResolvedAst.Root] {
 
     for {
       classes <- sequence(classesVal)
-      instances <- sequence(instancesVal) // MATT add synth instances
+      instances <- sequence(instancesVal)
       definitions <- sequence(definitionsVal)
       enums <- sequence(enumsVal)
       latticeComponents <- sequence(latticeComponentsVal)
       properties <- propertiesVal
     } yield ResolvedAst.Root(
-      classes.toMap, instances.toMap, definitions.toMap, enums.toMap, latticeComponents.toMap, properties.flatten, root.reachable, root.sources
+      classes.toMap, flatGroupToMap(instances), definitions.toMap, enums.toMap, latticeComponents.toMap, properties.flatten, root.reachable, root.sources
     )
   }
 
+  /**
+    * Creates a map from a list of key-(value list) pairs, appending in the case of duplicates.
+    */
+  private def flatGroupToMap[K, V](list: List[(K, List[V])]): Map[K, List[V]] = {
```

Comment:
Why not `toMap`? Is it combining the lists?

Reply:
toMap clobbers the results in the case of duplicate keys. Since instances of the same class can show up in different namespaces, we need to combine them more delicately by appending them together.

Reply:
Roger, so its the Monoid.combine you want :D (which we dont have in Scala)

Reply:
Can you call it `combine`?

Reply:
Yes, I must have been tired when I named that.